ARCHITECTURE.md

```diff
@@ -0,0 +1,38 @@
+# Skate Architecture
+
+## Overview
+
+Skate is a web app designed first (but not exclusively) for mobile. It uses Elixir and Phoenix on the server, and a single page React app on the client.
+
+## Data Flow Diagram
+
+![Skate data flow diagram](/documentation/images/skate-data-flow.png)
+
+## Data Sources
+
+Skate is read-only, so it only needs to ingest data and present it. Skate uses two main categories of data: schedule and real-time. Schedule data does not change frequently, so it is ingested once when the app loads, and persists in memory for the duration of the server instance. Real-time data, on the other hand, changes frequently, so these sources are queried at regular intervals.
+
+### Sources of Schedule Data
+
+- GTFS: A public, (mostly) standard data feed. Includes data like routes, stops, and trips.
+- HASTUS: A non-standard format export of private data. Includes operator and vehicle schedules.
+
+### Sources of Realtime Data
+
+- Busloc: GTFS-rt data
+  - Vehicle updates
+    - Driver information
+    - TransitMaster route assignments
+  - Trip updates
+    - Block waivers
+- Swiftly: GTFS-rt data
+  - Vehicle updates
+    - Inferred location information (what stop is the bus approaching)
+    - Schedule adherence (On-Time Performance)
+    - Headway
+    - "Predicted" route assignments based on location
```

Comment:
I don't think we do anything with predictions from Swiftly. We get trip assignments, but just for the current time. (It may be worth noting that swiftly's trip assignments are generally better than TransitMaster's).